flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSource.java

```diff
@@ -47,7 +55,8 @@
  * Flink Iceberg table source.
  */
 public class IcebergTableSource
-    implements ScanTableSource, SupportsProjectionPushDown, SupportsFilterPushDown, SupportsLimitPushDown {
+    implements ScanTableSource, LookupTableSource, SupportsProjectionPushDown, SupportsFilterPushDown,
```

Comment:
It is probably common to load Iceberg as `ScanTableSource`.  But it is unclear how Iceberg fits as a lookup source. @hailin0 can you provide some context on the actual use cases?

I thought LookupTableSource is meant for external storage that supports fast point query (like HBase/JDBC). `HBaseDynamicTableSource` also supports async lookup function.

Here is the Javadoc from `LookupTableSource`
```
 * A {@link DynamicTableSource} that looks up rows of an external storage system by one or more keys
 * during runtime.
 *
 * <p>Compared to {@link ScanTableSource}, the source does not have to read the entire table and can
 * lazily fetch individual values from a (possibly continuously changing) external table when
 * necessary.
```